### PR TITLE
Kh cted 1787

### PR DESCRIPTION
**Pull request by kurvinh**

### CODE REVIEW

File: classes/renderables/topic_render.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+class topic_render {
+    // Should always follow default button class of css theme.
+    const BUTTON_CLASS = 'btn btn-primary';
+
+    /**
+     * topic_subcription_button
+     * Builds new subscription button html desktop/mobile.
+     * @param string $last_reply_html
+     * @param null $currently_subbed
```
Comment (JJSwanevelder) [inline classes/renderables/topic_render.php:21]: Please add description to what the params are

File: classes/renderables/topic_render.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                                    <button type="button" class="trigger-subscribe rounded-pill ' . self::BUTTON_CLASS . '">'
+                                . get_string('topicfollowing','hsuforum') .
+                                '</button>
+                              </div>
+                              <div class="d-none d-sm-block d-md-none mobile-btn subscriber-wrapper">
+                                 <div class="last-reply-block">' . $last_reply_html . '</div>
+                                    <button  type="button" class="trigger-subscribe rounded-pill ' . self::BUTTON_CLASS . '">'
+                                    . get_string('topicfollowing','hsuforum') .
+                                    '</button>
+                              </div>';
```
Comment (JJSwanevelder) [inline classes/renderables/topic_render.php:39]: Is is possible to use templates within this function?

File: classes/renderables/topic_render.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        return $button_group;
+    }
+
+    /**
+     * contributors_html
+     * Takes a llist of users, and builds the avatar html based on settings.
+     * @param $users
+     * @return string
```
Comment (JJSwanevelder) [inline classes/renderables/topic_render.php:62]: Please add descriptive text on params

File: classes/renderables/topic_render.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public function contributors_html($users) {
+        $participants = '';
+        $avatar_list = '';
+        $avatars = implode(' ', $users->replyavatars);
+
+        $config = get_config('hsuforum');
+
+        if(!empty($config->avatarnumberstorenders)) {
```
Comment (JJSwanevelder) [inline classes/renderables/topic_render.php:71]: Add same !empty check as on line 74

File: renderer.php [hunk 5]
```diff
@@ -621,25 +620,22 @@ public function discussion_template($d, $forumtype) {
...
-
         $threadtitle = $d->subject;
         if (!$d->fullthread) {
             $threadtitle = "<a class='disable-router' href='$d->viewurl'>$d->subject</a>";
         }
         $options = get_string('options', 'hsuforum');
 
         //Add users country flag and timezone to the output.
+        // TODO: Rendering needs improving move button render here.
+        $topic_render = new Topic_render();
```
Comment (JJSwanevelder) [inline renderer.php:631]: Please adhere to moodle naming conventions here.

File: renderer.php [hunk 6]
```diff
@@ -1333,24 +1343,31 @@ public function toggle_element($type, $describedby, $url, $pressed = false, $lin
...
-        $icon = '<svg viewBox="0 0 100 100" class="svg-icon '.$type.'">
-        <title>'.$label.'</title>
-        <use xlink:href="#'.$type.'"></use></svg>';
+
+        $latestpost = '';
+        if (!empty($discussion->timemodified) && !empty($discussion->replies)) {
+            $latestpost = '<small class="hsuforum-thread-replies-meta">'.get_string('lastposttimeago', 'hsuforum', hsuforum_relative_time($discussion->timemodified)).'</small>';
+        }
+
+        $topic_render = new Topic_render();
```
Comment (JJSwanevelder) [inline renderer.php:1359]: Same naming convention here.

File: renderer.php [hunk 7]
```diff
@@ -2147,7 +2166,7 @@ public function discussion_navigation($prevdiscussion, $nextdiscussion) {
      *
      * @return string
      */
-     //fill="#FFFFFF"
+    //fill="#FFFFFF"
```
Comment (JJSwanevelder) [inline renderer.php:2169]: Remove commented out code.

**Review by JJSwanevelder — CHANGES_REQUESTED** [review]
Some initial comments

File: classes/renderables/topic_render.php (hunk 4 above)
Comment (kurvinh) [reply classes/renderables/topic_render.php:71]: not sure what you mean

File: classes/renderables/topic_render.php (hunk 2 above)
Comment (kurvinh) [reply classes/renderables/topic_render.php:39]: I guess it would be possible would you like me to make it a template?

**Review by JJSwanevelder — APPROVED** [review]
👍

**Pull request merged**